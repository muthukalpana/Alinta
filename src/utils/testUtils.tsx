import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { customerReducer } from "../redux/reducers/customerReducers";

// @TODO clean up types here
// Set-up copied from https://redux.js.org/recipes/writing-tests#components
function render(
  // @ts-ignore
  ui,
  {
    // @ts-ignore
    initialState,
    store = createStore(customerReducer, initialState),
    ...renderOptions
  } = {}
) {
  // @ts-ignore
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  // @ts-ignore
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
