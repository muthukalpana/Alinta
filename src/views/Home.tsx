import * as React from "react";
import { useSelector, shallowEqual, useDispatch } from "react-redux";
import { Customer } from "../components/Customer/Customer";
import { AddCustomerForm } from "../components/AddCustomer/AddCustomerForm";
import { Dispatch } from "redux";
import { CustomerState, ICustomer } from "../types/types";
import { addCustomer, removeCustomer } from "../redux/actions/customerActions";
import { FiChevronUp, FiChevronDown} from "react-icons/fi"
import {
  StyledHome,
  AccordionRow,
  AccordionLabel,
} from "./StyledHome";
import { useState } from "react";

interface State {
  showAddCustomer: boolean;
  showCustomers: boolean;
}

const Home: React.FC = () => {
  const customers: readonly ICustomer[] = useSelector(
    (state: CustomerState) => state.customers,
    shallowEqual
  );

  const [state, setState] = useState<State>({
    showAddCustomer: false,
    showCustomers: false,
  })

  const {
    showAddCustomer,
    showCustomers,
  } = state

  const dispatch: Dispatch<any> = useDispatch();

  const saveCustomer = React.useCallback(
    (customer: ICustomer) => dispatch(addCustomer(customer)),
    [dispatch]
  );

  return (
    <>
      <AccordionRow onClick={() => setState({
        ...state,
        showAddCustomer: !showAddCustomer,
      })}>
        <AccordionLabel> Add Customer </AccordionLabel>
        { showAddCustomer ? <FiChevronUp /> : <FiChevronDown /> }
      </AccordionRow>
      {showAddCustomer &&
        <AddCustomerForm
          customers={customers}
          saveCustomer={saveCustomer}
        />
      }
      <AccordionRow onClick={() => setState({
        ...state,
        showCustomers: !showCustomers,
      })}>
        <AccordionLabel> Customers </AccordionLabel>
        { showCustomers ? <FiChevronUp /> : <FiChevronDown /> }
      </AccordionRow>
      { showCustomers &&
        <StyledHome>
          {customers.map((customer: ICustomer) => (
            
              <Customer
                key={customer.id}
                customer={customer}
                removeCustomer={removeCustomer}
              />
          ))}
        </StyledHome>
      }
    </>
  );
};

export default Home;
