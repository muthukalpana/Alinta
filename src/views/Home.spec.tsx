import * as React from "react";
import { render } from  "../utils/testUtils";
import Home from "./Home";

describe("<Home />", () => {
  it("should render a <Home />", () => {
    const wrapper = render(<Home />);
    expect(wrapper.container).toMatchSnapshot();
  });

});
