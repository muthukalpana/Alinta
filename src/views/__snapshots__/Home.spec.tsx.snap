// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render a <Home /> 1`] = `
<div>
  <div
    class="sc-jrAFXE gfIxsg"
  >
    <div
      class="sc-kEjbQP ilsxlf"
    >
       Add Customer 
    </div>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
  <div
    class="sc-jrAFXE gfIxsg"
  >
    <div
      class="sc-kEjbQP ilsxlf"
    >
       Customers 
    </div>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
</div>
`;
